packages/earljs/src/matchers/ArrayWith.ts

```diff
@@ -0,0 +1,27 @@
+import { smartEq } from '../validators/smartEq'
+import { Matcher } from './Base'
+
+/**
+ * Matches an array containing the given item.
+ */
+export class ArrayWithMatcher<T> extends Matcher {
+  constructor(private readonly items: ReadonlyArray<T>) {
+    super()
+  }
+
+  check(v: unknown): boolean {
+    if (!Array.isArray(v)) {
+      return false
+    }
+
+    return v.some((i) => this.items.some((item) => smartEq(i, item).result === 'success'))
+  }
+
+  toString() {
+    return `[ArrayWith: ${this.items}]`
+  }
+
+  static make(...items: any[]): any {
```

Comment:
This can be typed as:

```
static make<T>(...items: T[]): T[
```

The reason why `containerWith` can't be typed properly and thus uses `any` is that it can be either set or array or any other iterable. And we can't type it simply as iterable b/c `toEqual` requires exact type assignability :/